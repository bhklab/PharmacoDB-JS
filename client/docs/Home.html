<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Home</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Home</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Home<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="Home"><span class="type-signature"></span>new Home<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Parent component for the home page,
includes child components for the description and stats.</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Home_Home.js.html">components/Home/Home.js</a>, <a href="components_Home_Home.js.html#line17">line 17</a>
    </li></ul></dd>
    

    

    

    
</dl>



















    <h5>Example</h5>
    
    <pre class="prettyprint"><code>return (
  &lt;Home/>
)</code></pre>



    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Compounds.html">Compounds</a></li><li><a href="Description.html">Description</a></li><li><a href="Home.html">Home</a></li><li><a href="Layout.html">Layout</a></li><li><a href="SearchHeader.html">SearchHeader</a></li><li><a href="Stats.html">Stats</a></li></ul><h3>Global</h3><ul><li><a href="global.html#getCompoundsQuery">getCompoundsQuery</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.4</a> on Fri Jun 26 2020 19:15:22 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>