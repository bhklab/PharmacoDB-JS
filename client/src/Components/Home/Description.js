import React from 'react';
import styled from 'styled-components';

import doseImg from '../../images/desc-dosecurves.png';
import colors from '../../styles/colors';

const StyledDescription = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;

    width: 70%;
    margin-top: 15vh;

    .text-container {
        width: 65%;

        display: flex;
        flex-direction: column;

        span {
            font-size: calc(0.5vw + 0.7em);
            line-height: calc(1vw + 1em);
        }

        h1 {
            color: ${colors.dark_teal_heading};
            font-family: 'Roboto Slab', serif;
            font-size: calc(2vw + 1em) !important;
            margin-bottom: 4vh;
        }
    }

    .dose-img {
        width: 30%;
        max-width: 400px;
    }
`;

/**
 * Shows the description on the home page
 * and the dose-response-curves graphic.
 *
 * @component
 * @example
 *
 * return (
 *   <Description/>
 * )
 */
const Description = () => {
    return (
        <StyledDescription>
            <div className="text-container">
                <h1>
                    Mine multiple cancer <br />
                    pharmacogenomic datasets.
                </h1>
                <span>
                    PharmacoDB allows scientists to search across publicly
                    available datasets to find instances where a drug or cell
                    line of interest has been profiled, and to view and compare
                    the dose-response data for a specific cell line - drug pair
                    from any of the studies included in the database.
                </span>
            </div>
            <img className="dose-img" src={doseImg} />
        </StyledDescription>
    );
};

export default Description;
