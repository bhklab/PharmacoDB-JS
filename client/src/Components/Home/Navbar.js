import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import logo from '../../images/pharmacodb-logo-dark.png';
import colors from '../../styles/colors';

const StyledNavbar = styled.div`
    position:absolute;
    width:100%;
    padding-top:40px;

    display:flex;
    justify-content: center;
    
    .container {
        width:70%;
        display:flex;
        justify-content:space-between;
        align-items: center;
    }

    .logo {
        width:200px;
    }
`;

const StyledLinks = styled.div`
    width:30%;
    display:flex;
    justify-content: space-between;
    align-items:center;

    a {
        color: ${colors.light_blue_header};
        font-family: 'Rubik', sans-serif;
        font-weight: 400;
        font-size: calc(0.4vw + 0.9em);
        letter-spacing:0.5px;
    }
`;

const Navbar = () => {
    const temp = "";
    return (
        <StyledNavbar>
            <div className="container">
                <img className="logo" src={logo}/>
                <StyledLinks>
                    <Link to="/">About</Link>
                    <Link to="/">Tools</Link>
                    <Link to="/">Datatypes</Link>
                </StyledLinks>
            </div>
        </StyledNavbar>
    );
};


export default Navbar;