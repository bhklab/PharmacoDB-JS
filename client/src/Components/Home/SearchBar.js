import React from 'react';
import styled from 'styled-components';

import colors from '../../styles/colors';

const TempBar = styled.div`
    width: 100%;
    background: rgb(233, 245, 255, 0.8);
    border-radius: 40px;
    height: 70px;
`;

const SearchBar = () => {
    return <TempBar />;
};

export default SearchBar;
