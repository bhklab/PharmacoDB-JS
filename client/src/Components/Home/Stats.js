import React from 'react';
import styled from 'styled-components';

import compoundsImg from '../../images/compounds.png';
import cellsImg from '../../images/cells.png';
import datasetsImg from '../../images/datasets.png';
import experimentsImg from '../../images/experiments.png';
import genesImg from '../../images/genes.png';
import tissuesImg from '../../images/tissues.png';
import colors from '../../styles/colors';

const StyledStats = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-evenly;

    width: 80%;
    margin-bottom: 20vh; //TODO: REMOVE
    margin-top: 8vh;
    background: ${colors.light_blue_bg};
    height: 14vw;
    border-radius: calc(0.8vw + 0.4em);

    .item-container {
        // width:calc(5.8vw + 40px);
        // height:calc(9vw + 30px);
        // border-radius:calc(1vw + 0.5em);
        background: ${colors.light_blue_bg};

        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;

        img {
            width: calc(4vw + 35px);
            margin-bottom: 1vw;
        }
        .text {
            color: ${colors.dark_teal_heading};
            font-size: calc(0.8vw + 0.3em);
            text-align: center;
            font-family: 'Rubik', sans-serif;
            font-weight: 400;
            line-height: calc(0.8vw + 0.4em);
        }
        .big {
            font-size: calc(0.9vw + 0.5em);
        }
    }
`;

/**
 * @returns {Object} - returns an Object of different types with name and value.
 */
const statistics = {
  datasets: { name: 'datasets', value: '7', image: datasetsImg },
  tissues: { name: 'tissues', value: '41', image: tissuesImg },
  cells: { name: 'cell-lines', value: '1,691', image: cellsImg },
  experiments: {
    name: 'experiments',
    value: '650,894',
    image: experimentsImg,
  },
  genes: { name: 'genes', value: '19,933', image: genesImg },
  compounds: { name: 'compounds', value: '759', image: compoundsImg },
};

/**
 * Shows the stats with graphics on the front page.
 *
 * @component
 * @example
 *
 * return (
 *   <Stats/>
 * )
 */
const Stats = () => (
  <StyledStats>
    {Object.keys(statistics).map((type) => (
      <div key={statistics[type].name} className="item-container">
        <img alt={statistics[type].value} src={`${statistics[type].image}`} />
        <div className="text">
          <span className="big">
            {`${statistics[type].value}`}
          </span>
          <br />
          {`${statistics[type].name}`}
        </div>
      </div>
    ))}
  </StyledStats>
);

export default Stats;
