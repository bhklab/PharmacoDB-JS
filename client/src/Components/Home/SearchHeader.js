import React from 'react';
import Navbar from './Navbar';
import SearchBar from './SearchBar';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import bg from '../../images/bg.jpg';
import colors from '../../styles/colors';

const StyledSearchHeader = styled.div`
    height: 45vh;
    background: ${colors.light_blue_header};
    background: url('${bg}');
    background-size: cover;
    background-attachment: fixed;
    background-position: center;
    
    display:flex;
    flex-direction:column;

    .search-container {
        width: 70%;
        margin: auto;

        h1 {
            font-family: 'Overpass', sans-serif;
            font-weight: 400;
            color: ${colors.light_blue_header};
            margin-bottom:20px;
        }
    }

    
`;
/**
 * Header component including the navbar and the
 * search bar. Is full size on home page, and minimized
 * on any other page (based on the page prop).
 * 
 * @component
 * @example
 * 
 * const page = "home"
 * return (
 *   <SearchHeader page={page}/>
 * )
 */
const SearchHeader = (props) => {
    const { page } = props; // check if page is empty and === "home"
    return (
        <StyledSearchHeader>
            <Navbar/>
            <div className="search-container">
                <h1>Try searching for a...</h1>
                <SearchBar/>
            </div>
        </StyledSearchHeader>
    )
};

SearchHeader.propTypes = {
    /**
     * SearchHeader's page name
     */
    page: PropTypes.string.isRequired,
};
  
SearchHeader.defaultProps = {
    page: ''
};

export default SearchHeader;