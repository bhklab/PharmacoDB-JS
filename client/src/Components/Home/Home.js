import React from 'react';
import styled from 'styled-components';
import ScrollToTop from 'react-scroll-to-top';
import Description from './Description';
import Layout from '../Utils/Layout';
import Stats from './Stats';
import 'react-scroll-to-top/lib/index.css';

const StyledHome = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

/**
 * Parent component for the home page,
 * includes child components for the description and stats.
 *
 * @component
 * @example
 *
 * return (
 *   <Home/>
 * )
 */
const Home = () => (
  <Layout page="home">
    <ScrollToTop smooth />
    <StyledHome>
      <Description />
      <Stats />
    </StyledHome>
  </Layout>
);

export default Home;
