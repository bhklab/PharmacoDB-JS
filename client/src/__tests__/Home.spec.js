import React from 'react';
import { mount } from 'enzyme';
import Home from '../Components/Home/Home';

// Catch-all snapshot test for entire rendering of home
it('renders correctly', () => {
  const component = mount(
    <Home />,
  );
  expect(component).toMatchSnapshot();
});
