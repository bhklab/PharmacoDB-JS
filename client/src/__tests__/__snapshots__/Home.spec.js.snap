// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Home />
</Router>
`;
